cluster/store.go

```diff
@@ -179,13 +179,15 @@ type Store struct {
 	lastAppliedIndexToDB atomic.Uint64
 	// / lastAppliedIndex index of latest update to the store
 	lastAppliedIndex atomic.Uint64
+
+	stateSetRaft func(raft *raft.Raft)
```

Comment:
can you add a godoc here explaining why we need that function and how it's used ?